devices/Button/GpioButton.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Device.Gpio;
+
+namespace Iot.Device.Button
+{
+    /// <summary>
+    /// GPIO implementation of Button.
+    /// Inherits from ButtonBase.
+    /// </summary>
+    public class GpioButton : ButtonBase
+    {
+        private const int DEFAULT_BUTTON_PIN = 37;
+
+        private GpioController _gpioController;
+        private int _buttonPin;
+        private bool _pullUp;
+
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Initialization of the button.
+        /// </summary>
+        /// <param name="buttonPin">GPIO pin of the button.</param>
+        /// <param name="pullUp">If the system is pullup (false = pulldown).</param>
+        /// <param name="doublePressMs">Max ticks between button presses to count as doublepress.</param>
+        /// <param name="holdingMs">Min ms a button is pressed to count as holding.</param>
+        public GpioButton(int buttonPin = DEFAULT_BUTTON_PIN, bool pullUp = true, int doublePressMs = DefaultDoublePressTicks, int holdingMs = DefaultHoldingMilliseconds)
```

Comment:
And also, I was quite sure I added it: please use TimeSpan for any time related elements, not int, not milliseconds. Internally, that can still be int but for external purpose, better with TimeSpan

Reply:
yes, it's still open in the comments.

Reply:
fixed